### ISSUE TITLE
Heading levels skipped

### ISSUE THREAD

**Issue by kimallen** [cisagov/manage.get.gov#3846]
### Current Behavior

There are some places where we skip heading levels. For example, we go from `<h1>` to `<h3>` and there is no `<h2>`. This is an accessibility violation. (https://www.w3.org/WAI/tutorials/page-structure/headings/).

### Expected Behavior

We should not skip heading levels

### Steps to Reproduce

Example incidence:
`portfolio_member_domains_edit.html`. To get here, go to edit domain assignments page.

<img width="1056" alt="Image" src="https://github.com/user-attachments/assets/38e64b75-d92b-4c2a-b058-d3031f7fefa1" />

We should do a full audit to ensure no heading levels are skipped throughout the application.
